src/mbgl/util/thread_pool.hpp

```diff
@@ -18,11 +38,8 @@ class ThreadPool final : public Scheduler {
     void schedule(std::function<void()>) override;
 
 private:
-    std::vector<std::thread> threads;
-    std::queue<std::function<void()>> queue;
-    std::mutex mutex;
-    std::condition_variable cv;
-    bool terminate{ false };
+    std::shared_ptr<LightweightThread::Impl> sharedImpl;
```

Comment:
This is difficult to read: ThreadPool responsibility moved to single shared LightweightThread::Impl while ThreadPool has a vector of LightweightThread.